server/plugin/webhook.go

```diff
@@ -327,67 +327,94 @@ func (p *Plugin) postPullRequestEvent(event *github.PullRequestEvent) {
 
 	newPRMessage, err := renderTemplate("newPR", event)
 	if err != nil {
-		p.API.LogWarn("Failed to render template", "error", err.Error())
+		p.API.LogWarn("Failed to render newPRCollapsed template", "error", err.Error())
+		return
+	}
+
+	newPRCollapsedMessage, err := renderTemplate("newPRCollapsed", event)
+	if err != nil {
+		p.API.LogWarn("Failed to render newPRCollapsed template", "error", err.Error())
 		return
 	}
 
+	pullRequestLabelledMessage := ""
+	pullRequestLabelledCollapsedMessage := ""
+	if action == "labeled" {
+		pullRequestLabelledMessage, err = renderTemplate("pullRequestLabelled", event)
+		if err != nil {
+			p.API.LogWarn("Failed to render template", "error", err.Error())
+			return
+		}
+
+		pullRequestLabelledCollapsedMessage, err = renderTemplate("pullRequestLabelledCollapsed", event)
+		if err != nil {
+			p.API.LogWarn("Failed to render template", "error", err.Error())
+			return
+		}
+	}
+
 	closedPRMessage, err := renderTemplate("closedPR", event)
 	if err != nil {
-		p.API.LogWarn("Failed to render template", "error", err.Error())
+		p.API.LogWarn("Failed to render closedPR template", "error", err.Error())
 		return
 	}
 
 	post := &model.Post{
-		UserId: p.BotUserID,
-		Type:   "custom_git_pr",
+		UserId:  p.BotUserID,
+		Type:    "custom_git_pr",
+		Message: "",
 	}
 
 	for _, sub := range subs {
 		if !sub.Pulls() && !sub.PullsMerged() {
 			continue
 		}
 
-		if sub.PullsMerged() && action != actionClosed {
+		if sub.PullsMerged() && action != "closed" {
 			continue
 		}
 
 		if p.excludeConfigOrgMember(event.GetSender(), sub) {
 			continue
 		}
 
-		label := sub.Label()
+		subscribedLabel := sub.Label()
 
 		contained := false
 		for _, v := range labels {
-			if v == label {
+			if v == subscribedLabel {
 				contained = true
 			}
 		}
 
-		if !contained && label != "" {
+		if !contained && subscribedLabel != "" {
 			continue
 		}
 
-		if action == actionLabeled {
-			if label != "" && label == eventLabel {
-				pullRequestLabelledMessage, err := renderTemplate("pullRequestLabelled", event)
-				if err != nil {
-					p.API.LogWarn("Failed to render template", "error", err.Error())
-					return
+		switch action {
+		case "labeled":
```

Comment:
This code should probably continue to use the namespaced `actionLabeled` constant etc.

```suggestion
		case actionLabeled:
```